### PR TITLE
[Feature] Add github action to connect with docker hub, add docker compose

### PR DESCRIPTION
**Pull request by MonsterChenzhuo**
### What problem does this PR solve?
[1.Streamx running on k8s
2.Quickly set up an experimental environment](https://github.com/streamxhub/streamx/issues/1008)
### What is changed and how it works?
1.add Dockerfile documents
2.add CI documents
3.add docker compose documents

### CODE REVIEW

File: .github/workflows/maven.yml [hunk 1]
```diff
@@ -7,18 +7,66 @@ on:
...
+          echo '======= Start building the database image ======='
+          echo '======= Start building streamx-mysql ======='
+          docker build -f docker/mysql/Dockerfile -t 18301455973/streamx-mysql:latest .
+
+      - name: push Images to DockerHub
+        run: |
+          docker images
+
+          echo '======= Start uploading streamx-server ======='
+          docker push 18301455973/streamx-server:latest
```
Comment (Whojohn) [inline .github/workflows/maven.yml]: Maybe special version from maven is better.

从 maven 依赖提取特定的版本号可能更好。

File: .github/workflows/maven.yml [hunk 2]
```diff
@@ -7,18 +7,55 @@ on:
...
+      - name: Maven Install
+        run: |
+          echo '======= Mixed front-end and back-end packaging ======='
+          mvn clean install -DskipTests -Dscala.version=${{ matrix.scala }} -Dscala.binary.version=2.12 -Pwebapp
 
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: 18301455973/streamx-server
```
Comment (wolfboys) [inline .github/workflows/maven.yml]: `18301455973` what is this? Is it an account

**Review by wolfboys — COMMENTED** [review]
LGTM.

**Comment by wolfboys** [comment]
@Al-assad @czy006  PTAL

File: streamx-console/streamx-console-service/src/assembly/plugins/streamx.yml [hunk 3]
```diff
@@ -0,0 +1,48 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: streamx
+  namespace: native-flink
```
Comment (czy006) [inline streamx-console/streamx-console-service/src/assembly/plugins/streamx.yml]: Maybe used default namespace is best to first deployed,this configuration let user to config it

File: .github/workflows/docker_push.yml [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_PASSWORD }}
+
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          images: streamxhub/streamx-server
```
Comment (wolfboys) [inline .github/workflows/docker_push.yml]: streamx-server need rename to streamx-console

File: docker-compose.yaml [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+version: '3.8'
+
+services:
+  streamx-server:
+    image: streamxhub/streamx-server:dev
```
Comment (wolfboys) [inline docker-compose.yaml]: streamx-server need rename to streamx-console

File: docker/server/Dockerfile-01 [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+#  limitations under the License.
+###############################################################################
+FROM maven:3.8.5-jdk-8 as deps-stage
+COPY . /
+WORKDIR /
+RUN tar zxvf streamx-console/streamx-console-service/target/streamx-console-service-*.tar.gz \
+&& mv streamx-console-service-* streamx
+
+
+FROM maven:3.8.4-jdk-8
```
Comment (czy006) [inline docker/server/Dockerfile-01]: FROM Maven Version is 3.8.5-jdk-8

File: docker/server/Dockerfile-02 [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+#  limitations under the License.
+###############################################################################
+FROM maven:3.8.5-jdk-8 as deps-stage
+COPY . /
+WORKDIR /
+RUN tar zxvf streamx-console/streamx-console-service/target/streamx-console-service-*.tar.gz \
+&& mv streamx-console-service-* streamx
+
+
+FROM maven:3.8.4-jdk-8
```
Comment (czy006) [inline docker/server/Dockerfile-02]: FROM Maven Version is 3.8.5-jdk-8

**Review by czy006 — COMMENTED** [review]
I have completed the review here, I will try to deploy it using docker compose later, if there is any problem I will report back

**Comment by czy006** [comment]
if use k8s deploy **Streamx**，maybe use **Helm3** to construction and build it  is the best choice to replace kubectl. This can be our goal in the future

**Comment by MonsterChenzhuo** [comment]
> 如果使用 k8s 部署 **Streamx **，也许使用 **Helm3 **构建和构建它是替代 kubectl 的最佳选择。 这可能是我们未来的目标

@czy006 I think this is a very good suggestion, then I want to modify it here and use helm for the installation, because I want to integrate Streamx services, Prometheus and Grafana and mysql all in a namespace called streamx.
What do you think?

**Review by wolfboys — APPROVED** [review]
LGTM.

**Comment by wolfboys** [comment]
@MonsterChenzhuo please rebase your pr. thx

**Pull request merged**